phase1/azure/do

```diff
@@ -67,13 +61,46 @@ check_auth() {
   fi
 }
 
+check_tenant_id() {
+  subscription_id="$(jq -r '.phase1.azure.subscription_id' ${configjson})"
+  tenant_id="$(curl https://management.azure.com/subscriptions/${subscription_id}?api-version=2016-01-01 -s -D - -o /dev/null | grep WWW-Authenticate | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)"
```

Comment:
Actually... that's not good. The http response is an error, so this breaks things.